### PR TITLE
modules: Remove unused code and variables

### PR DESCRIPTION
**Pull request by softwarecki**
Removed unnecessary functions from Processing Module Adapter. To ensure proper operation of native loadable modules it is necessary to bypass Processing Module Adapter used by FDK modules. This is currently done by overriding the pointer to module_interface used by the Module Adapter. Thanks to this, the Module Adapter directly calls functions provided by native module. As in this case the Processing Module Adapter functions are omitted, support for native libraries are removed from it as
it is no longer needed. This leads to remove is_native_sof variable and modules_process_raw, modules_process_audio_stream functions.

The code allocating/freeing in_buff and out_buff buffers, which were not used, was removed from the processing module adapter.

The value stored by the Processing Module Adapter in the module_adapter field of module_data structure has been moved to an unused private field. This change allowed the module_adapter field to be removed.

The unused sys_service field has been removed from the processing_module structure. It was never initialized anywhere, and its value was passed as a parameter to the native_system_agent_start function which did not use it.

The value assigned to the module_entry_point field in the module_data structure wasn't used anywhere. This field has been removed.

### CODE REVIEW

**Comment by dbaluta** [comment]
@softwarecki what is `native loadable modules`? Is the support from Zephyr for loading modules or the one from SOF?

**Comment by softwarecki** [comment]
> @softwarecki what is `native loadable modules`?
@dbaluta: 
We currently have three types of loadable modules...
1. FDK supported by Processing Module Aadapter,
2. native supported by lib_manager
3. llext supported by llext_manager and zephyr.

File: src/audio/module_adapter/module/modules.c
Comment (kv2019i) [inline src/audio/module_adapter/module/modules.c:67]: @softwarecki I don't think we have really explained "FDK" in SOF context, so following the git commit messages is a bit challenging. Correct Adrian me if I'm wrong, but in this context "FDK modules" is same as IADK modules which we have explained and documented in https://thesofproject.github.io/latest/architectures/firmware/intel/ace/iadk_modules.html .

So essentially, when you have native SOF loadable modules that don't depend on Intel IADK, the IADK specific code is not needed and code for native modules can be simplified like done in this PR.

I do find the language around "adapter"'s confusing. We have had adapters both between old SOF component.h interface and the new module interface. But we also have adapters between specialized interfaces (like Intel's IADK) and the module interface. This is not incorrect, but makes harder to follow changes like this PR.

**Review by kv2019i — COMMENTED** [review]
Seems ok. I'd appreciate if @pjdobrowolski , @lyakh  and others working with the loadable modules could review this before we go ahead and merge.

**Review by lyakh — DISMISSED** [review]

File: src/audio/module_adapter/module/modules.c
Comment (softwarecki) [reply src/audio/module_adapter/module/modules.c:67]: @kv2019i Yes, i mean IADK modules. FDK modules was rebranded to IADK.

@lyakh The `modules_init()` function is still used, called by Module Adapter. After detecting that it is not an IADK module, the pointer to `module_interface` is overwritten and points to the interface provided by a loaded module. In this way, all other functions of the Processing Module Adapter are omitted.

File: src/audio/module_adapter/module/modules.c
Comment (lyakh) [inline src/audio/module_adapter/module/modules.c]: this return code is now overwritten below without evaluation. Maybe at least add a print if it fails.

Comment (softwarecki) [reply src/audio/module_adapter/module/modules.c]: I will add a print. I think we would like free module code etc. regardless what its free method returns.

File: src/audio/module_adapter/module/modules.c
Comment (lyakh) [reply src/audio/module_adapter/module/modules.c:67]: @softwarecki but then - doesn't that mean, that `lib_manager_free_module()` is never called?

Comment (lyakh) [reply src/audio/module_adapter/module/modules.c:67]: ok, I think it is a problem, that operations are completely overwritten and `.free()` isn't called any more. But it also looks like that was introduced earlier by https://github.com/thesofproject/sof/commit/a0827526b7e1e4c88060812245a8cb9c381d5745 . I'll block this PR for now, until we clarify or fix this.

**Review by lyakh — CHANGES_REQUESTED** [review]

File: src/audio/module_adapter/module/modules.c
Comment (softwarecki) [reply src/audio/module_adapter/module/modules.c:67]: @lyakh: As you noticed, this PR does not introduce this change, so I see no reason to block it and thus postpone the next PR that will solve the described issue.

Comment (lyakh) [reply src/audio/module_adapter/module/modules.c:67]: @softwarecki well, I don't see a fix among currently submitted PRs and I think the offending commit should be reverted as per #8923 instead of being further deepened. Let's revert that commit and then we can base further work on modifying the shim on top of a working state.

**Review by pjdobrowolski — APPROVED** [review]

**Pull request merged**
